libyul/AsmAnalysis.cpp

```diff
@@ -625,7 +625,8 @@ void AsmAnalyzer::expectValidIdentifier(YulName _identifier, SourceLocation cons
 bool AsmAnalyzer::validateInstructions(std::string const& _instructionIdentifier, langutil::SourceLocation const& _location)
 {
 	// NOTE: This function uses the default EVM version instead of the currently selected one.
-	auto const builtin = EVMDialect::strictAssemblyForEVM(EVMVersion{}).builtin(YulName(_instructionIdentifier));
+	// TODO: Add EOF support
+	auto const builtin = EVMDialect::strictAssemblyForEVM(EVMVersion{}, std::nullopt).builtin(YulName(_instructionIdentifier));
 	if (builtin && builtin->instruction.has_value())
 		return validateInstructions(builtin->instruction.value(), _location);
```

Comment:
Let me know if you want this to be added in this PR

Reply:
Done. But we need test for this. Current `SyntaxTest` does not support `compileToEOF` flag. I would add it to `EVMVersionRestrictedTestCase` constructor to have it available in all derived TaseCase classes. WDYT? If so here on additional PR? :)

Reply:
It has to be done in this PR because CI requires to cover all possible error codes

Reply:
Either way sounds fine, but it's related so having it here is probably a good idea. And yeah, the error code - we have a whitelist for unused codes in `scripts/error_codes.py` so you could temporarily add it there if necessary, but that's always a last resort :)

Reply:
Though one thing to consider here - we have two different approaches to these settings, and you need to choose one:
1. `evmVersion` setting: you set a supported range of versions in the test, but a single isoltest run always executes only for one specific version (selected via `--evm-version`). If that matches the range, the test runs. Otherwise it's skipped. To get full coverage we run soltest with all relevant EVM versions in CI (via [`soltest_all.sh`](https://github.com/ethereum/solidity/blob/develop/.circleci/soltest_all.sh)).
1. `compileViaYul` setting in semantic tests: you select compilation via Yul or not or both. And each run does exactly that, running it twice if necessary. There's no CLI option.

The downside of (1) is that you'll have to run tests twice locally to test both EOF and legacy. It also adds another dimension to our test matrix. Currently we have two dimensions: optimizer on/off and EVM version. EOF version will be the third one, so you'll basically have to multiply the number of runs by two (though fortunately we can exclude older EVM versions from that, which lowers that number significantly).

Also, with (1) we end up unnecessarily running tests that are not affected by the flag multiple times, so it's only good for things that affect almost every part of the compiler.

The upside is that you get it supported almost for free in all test cases just by inheriting from `EVMVersionRestrictedTestCase`, while compiling things multiple times for (2) requires adding special support for it in each case. This matters especially for the smaller test cases (i.e. the ones where you used the flag from `CommonOptions`), where we definitely won't be going out of our way to make them use EOF, but it would be nice to have for more comprehensive coverage.

Since we already have that `--eof-version` flag in soltest and the option is closely related to EVM version, I think it still makes the most sense to go with (1). And we'll need to add that additional run with EOF enabled anyway, because otherwise the tests using EOF versions from `CommonOptions` now would still only be tested with legacy.

Reply:
So, in short, my suggestion would be to extend `EVMVersionRestrictedTestCase` to make it support a setting like `EOFVersion`, which takes a list of comma-separated values: for now only `legacy` and `>=1`. The values should really be ranges, like for EVM, but since we only have one version, I'd not bother with that - I'd just make it support `>=1` as a single, hard-coded value. If we ever get EOF v2, it will be easy to extend it by just changing how the value is parsed, without having to update it in all the tests.

Reply:
We agreed to add EOF support for AsmAnalysis in a separated PR. TODO @rodiazet: Add reference here when PR created